### PR TITLE
[WIP] Simplify wrappedCompile

### PR DESCRIPTION
**Pull request by singh264**
Closes: #12496
Signed-off-by: Amarpreet Singh amarpreet1997@gmail.com

### CODE REVIEW

**Comment by singh264** [comment]
@zl-wang @babsingh requesting preliminary feedback

**Comment by babsingh** [comment]
I am not the correct person to review the JIT changes. @dsouzai Can you please review these changes or assign a JIT reviewer for this PR?

File: runtime/compiler/control/CompilationThread.cpp [hunk 1]
```diff
@@ -8101,6 +8101,110 @@ TR::CompilationInfoPerThreadBase::compile(J9VMThread * vmThread,
...
+         {
+         TR_VerboseLog::writeLineLocked(TR_Vlog_COMPFAIL, "%s cannot be translated", compilee->signature(compileParameters->trMemory()));
+         }
+      Trc_JIT_noAttemptToJit(vmThread, compilee->signature(compileParameters->trMemory()));
+
+      return true;
+   }
+
+void
+TR::CompilationInfoPerThreadBase::setCompiledMethodOptimzationPlan(TR::CompilationInfoPerThreadBase *compilationInfo, TR_ResolvedMethod *compilee, TR_J9VMBase *vm)
```
Comment (dsouzai) [inline runtime/compiler/control/CompilationThread.cpp:8158]: I think this is incorrectly named, it should be something along the lines of `addUpgradeHintInSCCIfNeeded`.

File: runtime/compiler/control/CompilationThread.cpp [hunk 2]
```diff
@@ -8101,6 +8101,110 @@ TR::CompilationInfoPerThreadBase::compile(J9VMThread * vmThread,
...
+         Trc_JIT_cacheFull(vmThread);
+      if (jitConfig->runtimeFlags & J9JIT_DATA_CACHE_FULL)
+         Trc_JIT_dataCacheFull(vmThread);
+      compilationInfo->_methodBeingCompiled->_compErrCode = compilationExcessiveSize;
+
+      return true;
+   }
+
+bool
+TR::CompilationInfoPerThreadBase::isCompilingRestrictedMethod(TR::CompilationInfoPerThreadBase *compilationInfo, TR_ResolvedMethod *compilee, CompileParameters *compileParameters, TR_FilterBST *&filterInfo)
```
Comment (dsouzai) [inline runtime/compiler/control/CompilationThread.cpp:8132]: I think just calling this `isRestrictedMethod` is sufficient.

File: runtime/compiler/control/CompilationThread.cpp [hunk 3]
```diff
@@ -8101,6 +8101,110 @@ TR::CompilationInfoPerThreadBase::compile(J9VMThread * vmThread,
...
+   {
+      TR_J9VMBase *vm = compileParameters->_vm;
+      J9VMThread *vmThread = compileParameters->_vmThread;
+
+      // JITServer: methodCanBeCompiled check should have been done on the client, skip it on the server.
+      if (compilationInfo->_methodBeingCompiled->isOutOfProcessCompReq() || compilationInfo->methodCanBeCompiled(compileParameters->trMemory(), vm, compilee, filterInfo))
```
Comment (dsouzai) [inline runtime/compiler/control/CompilationThread.cpp:8138]: Better to split these into two lines.

File: runtime/compiler/control/CompilationThread.cpp [hunk 4]
```diff
@@ -8101,6 +8101,110 @@ TR::CompilationInfoPerThreadBase::compile(J9VMThread * vmThread,
...
+         {
+            return false;
```
Comment (dsouzai) [inline runtime/compiler/control/CompilationThread.cpp:8140]: Formatting.

File: runtime/compiler/control/CompilationThread.cpp [hunk 5]
```diff
@@ -8101,6 +8101,110 @@ TR::CompilationInfoPerThreadBase::compile(J9VMThread * vmThread,
...
 
+bool
+TR::CompilationInfoPerThreadBase::isCompilingWithExcessiveCacheSize(TR::CompilationInfoPerThreadBase *compilationInfo, CompileParameters *compileParameters)
+   {
+      J9JITConfig *jitConfig = compilationInfo->_jitConfig;
+      J9VMThread *vmThread = compileParameters->_vmThread;
+
+      if (!(jitConfig->runtimeFlags & (J9JIT_CODE_CACHE_FULL | J9JIT_DATA_CACHE_FULL)))
+         {
+            return false;
```
Comment (dsouzai) [inline runtime/compiler/control/CompilationThread.cpp:8112]: Formatting.

File: runtime/compiler/control/CompilationThread.cpp [hunk 6]
```diff
@@ -8101,6 +8101,110 @@ TR::CompilationInfoPerThreadBase::compile(J9VMThread * vmThread,
    return startPC;
    }
 
+bool
+TR::CompilationInfoPerThreadBase::isCompilingWithExcessiveCacheSize(TR::CompilationInfoPerThreadBase *compilationInfo, CompileParameters *compileParameters)
```
Comment (dsouzai) [inline runtime/compiler/control/CompilationThread.cpp:8105]: A better name might be `isCodeOrDataCacheFull`.

File: runtime/compiler/control/CompilationThread.cpp [hunk 7]
```diff
@@ -8101,6 +8101,110 @@ TR::CompilationInfoPerThreadBase::compile(J9VMThread * vmThread,
...
+         compilee = vm->createResolvedMethod(compileParameters->trMemory(), method, NULL, (TR_OpaqueClassBlock *) niDetails.classNeedingThunk());
+         }
+      else
+         {
+         compilee = vm->createResolvedMethod(compileParameters->trMemory(), method);
+         }
+
+      setCompiledMethodOptimzationPlan(compilationInfo, compilee, vm);
+
+      if (isCompilingRestrictedMethod(compilationInfo, compilee, compileParameters, filterInfo) || isCompilingWithExcessiveCacheSize(compilationInfo, compileParameters))
```
Comment (dsouzai) [inline runtime/compiler/control/CompilationThread.cpp:8200]: Better to split this into two lines.

**Review by dsouzai — CHANGES_REQUESTED** [review]
Overall looks good; you're on the right track. Thanks for picking this up @singh264 :).

**Comment by singh264** [comment]
`method_name line_start - line_end`

createCompilee [8436 - 8469](https://github.com/singh264/openj9/blob/refactor_wrappedcompile_oct15/runtime/compiler/control/CompilationThread.cpp#L8436-L8469)
addUpgradeHintInSCCIfNeeded [8419 - 8434](https://github.com/singh264/openj9/blob/refactor_wrappedcompile_oct15/runtime/compiler/control/CompilationThread.cpp#L8419-L8434)
isRestrictedMethod [8392 - 8417](https://github.com/singh264/openj9/blob/refactor_wrappedcompile_oct15/runtime/compiler/control/CompilationThread.cpp#L8392-L8417)
isCodeOrDataCacheFull [8365 - 8390](https://github.com/singh264/openj9/blob/refactor_wrappedcompile_oct15/runtime/compiler/control/CompilationThread.cpp#L8365-L8390)
isOutOfProcessCompReq [8356 - 8363](https://github.com/singh264/openj9/blob/refactor_wrappedcompile_oct15/runtime/compiler/control/CompilationThread.cpp#L8356-L8363)
processOutOfProcessComp [8332 - 8354](https://github.com/singh264/openj9/blob/refactor_wrappedcompile_oct15/runtime/compiler/control/CompilationThread.cpp#L8332-L8354)
processNonOutOfProcessComp [8227 - 8330](https://github.com/singh264/openj9/blob/refactor_wrappedcompile_oct15/runtime/compiler/control/CompilationThread.cpp#L8227-L8330)
aotCompilationReUpgradedToWarm [7870 - 7897](https://github.com/singh264/openj9/blob/refactor_wrappedcompile_oct15/runtime/compiler/control/CompilationThread.cpp#L7870-L7897)
adjustOptionsForAotCompilation [7840 - 7868](https://github.com/singh264/openj9/blob/refactor_wrappedcompile_oct15/runtime/compiler/control/CompilationThread.cpp#L7840-L7868)
tweakAotLoadCompilationStrategy [8030 - 8225](https://github.com/singh264/openj9/blob/refactor_wrappedcompile_oct15/runtime/compiler/control/CompilationThread.cpp#L8030-L8225)
processSamplingJProfiling [7899 - 8028](https://github.com/singh264/openj9/blob/refactor_wrappedcompile_oct15/runtime/compiler/control/CompilationThread.cpp#L7899-L8028)

**Comment by singh264** [comment]
I think https://github.com/singh264/openj9/pull/2 is an extension of the changes in this pull request.

**Comment by ymanton** [comment]
This PR has merge conflicts. Please fix those before requesting a review.

**Pull request closed without merging**